actors/builtin/market/market_test.go

```diff
@@ -274,27 +271,119 @@ func TestMarketActor(t *testing.T) {
 
-			rt.SetCaller(worker, builtin.AccountActorCodeID)
-			params := market.WithdrawBalanceParams{
-				ProviderOrClientAddress: provider,
-				Amount:                  withdrawAmount,
-			}
+			rt.GetState(&st)
+			assert.Equal(t, abi.NewTokenAmount(0), st.GetEscrowBalance(rt, provider))
+		})
 
-			actor.expectProviderControlAddressesAndValidateCaller(rt, provider, owner, worker)
-			rt.ExpectSend(owner, builtin.MethodSend, nil, abi.NewTokenAmount(20), nil, exitcode.Ok)
+		// TODO: withdraws limited by slashing
 
-			rt.Call(actor.WithdrawBalance, &params)
+		t.Run("client balance after withdrawal must ALWAYS be greater than or equal to locked amount", func(t *testing.T) {
+			rt, actor := setup()
 
-			rt.Verify()
+			// create the deal to publish and the epochs around it
+			startEpoch := abi.ChainEpoch(10)
+			endEpoch := abi.ChainEpoch(20)
+			currentEpoch := abi.ChainEpoch(5)
+			deal := actor.generateUnVerifiedDealProposal(client, provider, startEpoch, endEpoch)
+
+			// ensure client and provider have enough funds to lock for the deal
+			actor.addParticipantFunds(rt, client, deal.ClientBalanceRequirement())
+			actor.addProviderFunds(rt, provider, owner, worker, deal.ProviderBalanceRequirement())
 
+			// publish the deal so that client collateral is locked
+			rt.SetEpoch(currentEpoch)
+			actor.publishDealOK(rt, deal, owner, worker, provider)
 			rt.GetState(&st)
-			assert.Equal(t, abi.NewTokenAmount(0), st.GetEscrowBalance(rt, provider))
+			require.Equal(t, deal.ClientBalanceRequirement(), st.GetLockedBalance(rt, client))
+
+			// client cannot withdraw any funds since all it's balance is locked
+			withDrawAmt := abi.NewTokenAmount(1)
+			actualWithdrawn := abi.NewTokenAmount(0)
+			actor.withdrawClientBalanceOK(rt, client, withDrawAmt, actualWithdrawn)
+
+			// add some more funds to the client & ensure withdrawal is limited by the locked funds
+			actor.addParticipantFunds(rt, client, abi.NewTokenAmount(25))
+			withDrawAmt = abi.NewTokenAmount(30)
+			actualWithdrawn = abi.NewTokenAmount(25)
+
+			actor.withdrawClientBalanceOK(rt, client, withDrawAmt, actualWithdrawn)
 		})
 
-		// TODO: withdraws limited by slashing
-		// TODO: withdraws limited by locked balance
-		// https://github.com/filecoin-project/specs-actors/issues/465
+		t.Run("worker balance after withdrawal must ALWAYS be greater than or equal to locked amount", func(t *testing.T) {
+			rt, actor := setup()
+
+			// create the deal to publish and the epochs around it
+			startEpoch := abi.ChainEpoch(10)
+			endEpoch := abi.ChainEpoch(20)
+			currentEpoch := abi.ChainEpoch(5)
+			deal := actor.generateUnVerifiedDealProposal(client, provider, startEpoch, endEpoch)
+
+			// ensure client and provider have enough funds to lock for the deal
+			actor.addParticipantFunds(rt, client, deal.ClientBalanceRequirement())
+			actor.addProviderFunds(rt, provider, owner, worker, deal.ProviderBalanceRequirement())
+
+			// publish the deal so that provider collateral is locked
```

Comment:
Since the deal always locks both sides, combine this with the above test. There's nothing distinct about the set-up between the two.